packages/core/snapshot/read.go

```diff
@@ -85,68 +78,129 @@ func readSnapshotHeader(reader io.ReadSeeker) (*ledger.SnapshotHeader, error) {
 	}
 	header.DiffEpochIndex = epoch.Index(index)
 
+	var latestECRecordLen int64
+	if err := binary.Read(reader, binary.LittleEndian, &latestECRecordLen); err != nil {
+		return nil, fmt.Errorf("unable to read latest ECRecord bytes len: %w", err)
+	}
+
+	ecRecordBytes := make([]byte, latestECRecordLen)
+	if err := binary.Read(reader, binary.LittleEndian, ecRecordBytes); err != nil {
+		return nil, fmt.Errorf("unable to read latest ECRecord: %w", err)
+	}
+	header.LatestECRecord = &epoch.ECRecord{}
+	if err := header.LatestECRecord.FromBytes(ecRecordBytes); err != nil {
+		return nil, err
+	}
+
 	return header, nil
 }
 
-func readSolidEntryPoints(scanner *bufio.Scanner) (seps *SolidEntryPoints, err error) {
-	scanner.Scan()
-	data := scanner.Bytes()
+func readSolidEntryPoints(reader io.ReadSeeker) (seps *SolidEntryPoints, err error) {
+	seps = &SolidEntryPoints{}
+	blkIDs := make([]tangleold.BlockID, 0)
+
+	// read seps EI
+	var index int64
+	if err := binary.Read(reader, binary.LittleEndian, &index); err != nil {
+		return nil, fmt.Errorf("unable to read fullEpochIndex: %w", err)
+	}
+	seps.EI = epoch.Index(index)
+
+	// read numbers of solid entry point
+	var sepsLen int64
+	if err := binary.Read(reader, binary.LittleEndian, &sepsLen); err != nil {
+		return nil, fmt.Errorf("unable to read seps len: %w", err)
+	}
+
+	spentLenInt := int(sepsLen)
+	for i := 0; i < spentLenInt; {
```

Comment:
I don't understand why we need this for loop: don't we have a single solidEntryPoint set of blocksIDs per epoch index we read from the snapshot file? We should know ahead of time how many EI + SEPS combos we should read.

Reply:
Here is to read blockIDs in chunks, we stop when we get the number of total seps (`sepsLenInt`) in the epoch. Doing this could avoid crashing the reader with a big slice of bytes at once, same goes for other data types.

The reading/writing process is:
1. write/read the total number of data types
2. split data into chunks (size = 100)
    a. write/read the bytes length of chunk
    b. write/read the data with fixed length, the reader will consume deserialized data 
    c. repeat a., b., until we hit the total number
3. move on to the next data type if any